pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -308,10 +310,17 @@ public void close() throws PulsarServerException {
         try {
             closeAsync().get();
         } catch (ExecutionException e) {
-            if (e.getCause() instanceof PulsarServerException) {
-                throw (PulsarServerException) e.getCause();
+            Throwable cause = e.getCause();
+            if (cause instanceof PulsarServerException) {
+                throw (PulsarServerException) cause;
+            } else if (cause instanceof TimeoutException) {
+                if (getConfiguration().getBrokerShutdownTimeoutMs() < 1000L) {
+                    // ignore shutdown timeout when it's less than 1000ms (in tests)
```

Comment:
The behavior should be consistent between the Pulsar server and the test? If users configure the shutdown timeout which < 1000L, what is the expected behavior.

Reply:
The reason to do this choice is to simplify the configuration for tests. This PR sets the brokerShutdownTimeMs to 100 ms in tests. If the timeout exception wouldn't be ignored, special handling would be needed in all tests to catch the TimeoutException and ignore it. I'm expecting that it's a justified tradeoff to have different behavior if the user goes and configured shutdown timeout to less than 1000 ms. WDYT?

Reply:
I guess one alternative is to use the timeout value of 0 in tests and ignore the timeout exceptions when timeout is 0. That could be less surprising than having the 1000ms boundary.

Reply:
But even if we ignore the timeout under 1000ms, the test still can occur the TimeoutException? Most tests are extends from the `MockedPulsarServiceBaseTest`, can we deal with this in the `MockedPulsarServiceBaseTest`?

Reply:
> I guess one alternative is to use the timeout value of 0 in tests and ignore the timeout exceptions when timeout is 0. That could be less surprising than having the 1000ms boundary

Sound good to me.

Reply:
I pushed changes to address this. @codelipenghui Please take another look.